pkg/engine/rule/r0008_malicious_ssh_connection.go

```diff
@@ -0,0 +1,183 @@
+}
+
+func CreateRuleR0008MaliciousSSHConnection() *R0008MaliciousSSHConnection {
+	return &R0008MaliciousSSHConnection{accessRelatedFiles: false,
+		sshInitiatorPid:           0,
+		configFileAccessTimeStamp: 0,
+		allowedPorts:              []uint16{22},
+	}
+}
+
+func (rule *R0008MaliciousSSHConnection) SetParameters(params map[string]interface{}) {
+	if params["allowedPorts"] != nil {
+		rule.allowedPorts = params["allowedPorts"].([]uint16)
+	}
+}
+
+func (rule *R0008MaliciousSSHConnection) DeleteRule() {
+}
+
+func (rule *R0008MaliciousSSHConnection) ProcessEvent(eventType tracing.EventType, event interface{}, appProfileAccess approfilecache.SingleApplicationProfileAccess, engineAccess EngineAccess) RuleFailure {
+	if eventType != tracing.OpenEventType && eventType != tracing.NetworkEventType {
+		return nil
+	}
+
+	if eventType == tracing.OpenEventType && !rule.accessRelatedFiles {
+		openEvent, ok := event.(*tracing.OpenEvent)
+		if !ok {
+			return nil
+		} else {
+			if IsSSHConfigFile(openEvent.PathName) {
+				rule.accessRelatedFiles = true
+				rule.sshInitiatorPid = openEvent.Pid
+				rule.configFileAccessTimeStamp = openEvent.Timestamp
+			}
+
+			return nil
+		}
+	} else if eventType == tracing.NetworkEventType && rule.accessRelatedFiles {
+		networkEvent, ok := event.(*tracing.NetworkEvent)
+		if !ok {
+			return nil
+		}
+
+		timestampDiffInSeconds := calculateTimestampDiffInSeconds(networkEvent.Timestamp, rule.configFileAccessTimeStamp)
+		if timestampDiffInSeconds > MaxTimeDiffInSeconds {
+			rule.accessRelatedFiles = false
+			rule.sshInitiatorPid = 0
+			rule.configFileAccessTimeStamp = 0
+			return nil
+		}
+		if networkEvent.Pid == rule.sshInitiatorPid && networkEvent.PacketType == "OUTGOING" && networkEvent.Protocol == "TCP" && !slices.Contains(rule.allowedPorts, networkEvent.Port) {
+			rule.accessRelatedFiles = false
+			rule.sshInitiatorPid = 0
+			rule.configFileAccessTimeStamp = 0
+			return &R0008MaliciousSSHConnectionFailure{
+				RuleName:         rule.Name(),
+				Err:              fmt.Sprintf("ssh connection to port %d is not allowed", networkEvent.Port),
+				FixSuggestionMsg: "If this is a legitimate action, please add the port as a parameter to the binding of this rule",
+				FailureEvent:     networkEvent,
+				RulePriority:     R0008MaliciousSSHConnectionRuleDescriptor.Priority,
+			}
+		}
+	}
+
+	return nil
+}
+
+func calculateTimestampDiffInSeconds(timestamp1 int64, timestamp2 int64) int64 {
+	return (timestamp1 - timestamp2) / int64(time.Second)
+}
+
+func IsSSHConfigFile(path string) bool {
+	for _, sshFile := range SSHRelatedFiles {
+		if strings.Contains(path, sshFile) {
+			log.Printf("Found SSH related file: %s\n", path)
+			return true
+		}
+	}
+	return false
+}
+
+func (rule *R0008MaliciousSSHConnection) Requirements() RuleRequirements {
+	return RuleRequirements{
+		EventTypes:             []tracing.EventType{tracing.OpenEventType, tracing.NetworkEventType},
+		NeedApplicationProfile: false,
```

Comment:
I think we should have a clear decision what should be in the app profile and what in the parameters.
This is a comment with no AI

Reply:
This rule doesn't need application profile because imo